dashboard/src/components/ui/StatusCircle.tsx

```diff
@@ -1,9 +1,11 @@
-import clsx from 'clsx';
+import { twMerge } from 'tailwind-merge';
 
 export type DeploymentStatus =
   | 'DEPLOYING'
   | 'DEPLOYED'
   | 'FAILED'
+  | 'PENDING'
```

Comment:
These three states match the `deploymentStatus` column in the `deployments` table:

<table><tbody><tr><td>DEPLOYING</td></tr><tr><td>DEPLOYED</td></tr><tr><td>FAILED</td></tr></tbody></table>

do the new ones 'PENDING' & 'SCHEDULED' are also new states in this column? @elephant3 I haven't seen them in staging, do we mutate a given deployment to go from, say, PENDING -> FAILED? I think we could also look up into creating these states as an enum-like similar to what we have with the `app_states`

Reply:
Yes, they are - https://github.com/nhost/watchtower/blob/ce804295c13a4d0067e20a941123ae2de77fb1ce/pkg/hasura/deployments_notificator.go#L18-L23

`PENDING` -> `FAILED` is possible, as well as `SCHEDULED` -> `FAILED` if they get stuck

Reply:
Ah, got it. Thanks for clarifying. I'll open up an issue in console-next in the event we want to have something like the \`**app\_states\`** table to get a grasp for all possible states of a deployment :+1:

Reply:
> Ah, got it. Thanks for clarifying. I'll open up an issue in console-next in the event we want to have something like the \`**app\_states\`** table to get a grasp for all possible states of a deployment :+1: